### PR TITLE
fix(influxdb_metrics sink): Use `i` for unsigned ints on Influx v1

### LINKED ISSUE #2867: Influx 1.0 sink produces invalid unsigned integers
Influx 1.0 does not support unsigned integers (`u`), but does support integers (`i`). Sending Histogram data over to Influx 1.0 responds back with a `400 Bad Request`, and the following error in the body and headers:

```
X-Influxdb-Error: unable to parse 'test.example_seconds,cost=12,metric_type=histogram,result=incorrect bucket_1=1u,bucket_0.025=0u,bucket_0.01=0u,bucket_0.05=0u,bucket_0.5=1u,bucket_0.1=0u,bucket_0.25=0u,bucket_2.5=1u,bucket_0.005=0u,bucket_10=1u,count=1,sum=0.322013824,bucket_5=1u 1592511668537779000': invalid number
X-Influxdb-Version: 1.7.10
```

Changing all of the `u`s to `i`s results in the expected `204 No Content` response.

One potential solution is to have all Unsigned Integers transmitted with `i` when the backend is a 1.0 Influx DB.

Sources:
* https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_tutorial/#data-types
* https://v2.docs.influxdata.com/v2.0/reference/syntax/line-protocol/

### PR DESCRIPTION
**Pull request by neoeinstein**
fixes #2867

### CODE REVIEW

**Comment by binarylogic** [comment]
@bednar would you mind taking a look at this? I'm curious what you think. Thanks!

File: src/sinks/influxdb/mod.rs [hunk 1]
```diff
@@ -174,6 +189,7 @@ fn healthcheck(
 
 // https://v2.docs.influxdata.com/v2.0/reference/syntax/line-protocol/
 fn influx_line_protocol(
+    protocol_version: ProtocolVersion,
```
Comment (bednar) [inline src/sinks/influxdb/mod.rs:192]: What about pass `protocol_version` as a reference?

```suggestion
    protocol_version: &ProtocolVersion,
```

**Review by bednar — COMMENTED** [review]
I think it's correct fix by @neoeinstein 👍.

There is also an issue  about supporting of `unsigned `int into 1.x, but it's still open https://github.com/influxdata/influxdb/issues/17781

File: src/sinks/influxdb/mod.rs (hunk 1 above)
Comment (fanatid) [reply src/sinks/influxdb/mod.rs:192]: Reference should not give any advantages for such simple enum

**Review by fanatid — APPROVED** [review]
Thank you!

**Review by bruceg — APPROVED** [review]
Looks good to me too, we'll wait for the additional requested test before merging.

File: src/sinks/influxdb/logs.rs [hunk 2]
```diff
@@ -223,13 +227,57 @@ mod tests {
...
+    }
+
     #[test]
     fn test_encode_event() {
         let mut event = Event::from("hello");
         event.as_mut_log().insert("host", "aws.cloud.eur");
         event.as_mut_log().insert("source_type", "file");
 
-        event.as_mut_log().insert("int", 4);
+        event.as_mut_log().insert("int", 4i32);
```
[the remark this answers is not in the document]
Comment (neoeinstein) [reply src/sinks/influxdb/logs.rs:280]: I had earlier added a test field with an unsigned int, but there is no [`Value` for an unsigned int](https://github.com/timberio/vector/blob/cdebd75c0c0d0648de8f96a4503d1df6128d07f8/src/event/mod.rs#L266). So, all values would be treated as signed integers in this context.

**Comment by bruceg** [comment]
Since generating that test will be more effort than it's worth (`Value` doesn't make a signed vs unsigned distinction), I'll just merge this as is.

**Pull request merged**